components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -4054,6 +4108,7 @@ def getUnionTypeTemplateVars(type, descriptorProvider):
         type, descriptorProvider, failureCode="return Ok(None);",
         exceptionCode='return Err(());',
         isDefinitelyObject=True)
+    assert not info.needsRooting
```

Comment:
We should support unions containing typed arrays; various network and webgl specs require them.